manager/dispatcher/dispatcher.go

```diff
@@ -116,6 +116,17 @@ func (b weightedPeerByNodeID) Len() int { return len(b) }
 
 func (b weightedPeerByNodeID) Swap(i, j int) { b[i], b[j] = b[j], b[i] }
 
+// tasksEqual returns true if the tasks are functionaly equal, ignoring status,
+// version and other superfluous fields.
+func tasksEqual(a, b *api.Task) bool {
+	a, b = a.Copy(), b.Copy()
```

Comment:
This should use shallow copy. 

```
	// shallow copy
 	copyA, copyB := *a, *b
```

See https://github.com/docker/swarmkit/commit/af58bcaf8515795e057410cedf7f255ab33ae74a